src/algorithms.jl

```diff
@@ -114,3 +114,37 @@ Base.@kwdef struct SumOfSquaresEnclosure{T} <: AbstractEnclosureAlgorithm
     backend::T
     order::Int = 5
 end
+
+"""
+    BranchAndBoundEnclosure
+
+Data type to bound the range of `f` over `X` using the branch and bound algorithm.
+
+### Fields
+
+- `maxdepth` (default `10`): maximum depth of the search tree
+- `tol` (default `1e-3`): tolerance to compute the range of the function
+
+### Algorithm
+
+It evaluates the function over the interval `X`. If the maximum depth is reached or the
+width of `f(X)` is below the tolerance, it returns the computed range, otherwise it bisects
```

Comment:
```suggestion
The algorithm evaluates a function `f` over an interval `X`. If the maximum depth is reached or the
width of `f(X)` is below the tolerance, the algorithm returns the computed range; otherwise it bisects
```